src/Containers/Microsoft.NET.Build.Containers/FallbackToHttpMessageHandler.cs

```diff
@@ -12,13 +12,16 @@ namespace Microsoft.NET.Build.Containers;
 /// </summary>
 internal sealed partial class FallbackToHttpMessageHandler : DelegatingHandler
 {
+    private readonly string _registeryName;
     private readonly string _host;
     private readonly int _port;
     private readonly ILogger _logger;
     private bool _fallbackToHttp;
 
-    public FallbackToHttpMessageHandler(string host, int port, HttpMessageHandler innerHandler, ILogger logger) : base(innerHandler)
+    public FallbackToHttpMessageHandler(string registryName, string host, int port, HttpMessageHandler innerHandler, ILogger logger)
+        : base(innerHandler)
```

Comment:
Minor suggestion. May be `FallbackToHttpMessageHandler` doesn't have to know about `registryName`. This could be done by using an `int fallbackPort` argument, that gets initialized based on `registryName` where this type is created.

```cs
// Fall back to port 80 if the default https port was assumed.
int fallbackPort = baseUri.Port = 443 && !registryName.EndsWith(":443", StringComparison.OrdinalIgnoreCase) : 80 : baseUri.Port;
new FallbackToHttpMessageHandler(registryName, baseUri.Host, baseUri.Port, fallbackPort, socketsHttpHandler, logger);
```

Reply:
Yes, It's maybe better, but not much. Thanks for you suggestion.

I am interesting about your PR abount the unit test about this feature, can you give me some link about this?

Reply:
https://github.com/dotnet/sdk/pull/43491/ does tests by faking a server.